spaceship/lib/spaceship/client.rb

```diff
@@ -124,8 +124,8 @@ def teams
     #  "userName"=>"detlef@krausefx.com"}
     def user_details_data
       return @_cached_user_details if @_cached_user_details
-      r = request(:get, '/WebObjects/iTunesConnect.woa/ra/user/detail')
-      @_cached_user_details = parse_response(r, 'data')
+      r = request(:get, "https://appstoreconnect.apple.com/olympus/v1/session")
```

Comment:
We already make this request here https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/client.rb#L562

Maybe have that use `user_details_data`?

Reply:
Ohhh, we do call this already! `user_details_data` caches responses and the `fetch_olympus_session` does not so I think we would need to do mesh them together some how 🤔 

I would like to combine them but I might punt on that to a future PR because I'm scared of making that change tonight 😅

Reply:
Indeed, we noticed that `fetch_olympus_session` also called this endpoint, but the authentication logic of fastlane is quite complex and we didn’t feel comfortable doing that change, at least not at this stage, as the implications of dealing with it and with caching weren’t immediately clear 🙏